types/webpack-dev-server/index.d.ts

```diff
@@ -70,20 +70,15 @@ declare namespace WebpackDevServer {
 
 declare class WebpackDevServer {
     constructor(
-        webpack: webpack.compiler.Compiler,
+        webpack: webpack.Compiler | webpack.MultiCompiler,
```

Comment:
Probably `webpack.ICompiler`?

Reply:
The suggestion is based on `webpack-dev-middleware`'s implementation and typing.

https://github.com/webpack/webpack-dev-server/blob/master/lib/Server.js#L82
https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/webpack-dev-middleware/index.d.ts#L12

Reply:
No.

In webpack it is declared as such;
```
declare function webpack(options?: webpack.Configuration): webpack.Compiler;
```

Reply:
In webpack it is declared as such;
```
declare function webpack(options?: webpack.Configuration): webpack.Compiler;
```

`Compiler` extends `ICompiler` and just to mention `ICompiler` should be renamed as it's not following Typescript guidelines. As interfaces shouldn't not be prefixed with `I`

Reply:
ICompiler is an interface. Compiler is a concrete class that implements the interface and extends Tapable. MultiComplier is an abstract class that implements the interface. If you check the implementation, the webpack dev server uses a method named plugin which is from Tapable which is inherited by Compiler only. MultiComplier doesn't guarantee to provide this method. I feel Compiler should be the best match. Not even the ICompiler I suggested... Again, this is a suggestion.

As of the guideline, it's for The Typescript project itself. Personally, I think allowing prefix 'I' to interface name is Ok. There are discussions about this guideline out there, such as https://stackoverflow.com/questions/31876947/confused-about-the-interface-and-class-coding-guidelines-for-typescript. I've been using TS for several projects and found that simple prefix like 'I' is helpful in identifying the significant code quickly. 🙂

Reply:
So `PR` approved :P?

Reply:
yes